mobile/native/src/cipher.rs

```diff
@@ -0,0 +1,112 @@
+use aes_gcm_siv::AeadInPlace;
+use aes_gcm_siv::Aes256GcmSiv;
+use aes_gcm_siv::KeyInit;
+use aes_gcm_siv::Nonce;
+use anyhow::anyhow;
+use anyhow::Result;
+use bitcoin::secp256k1::ecdsa::Signature;
+use bitcoin::secp256k1::rand;
+use bitcoin::secp256k1::rand::Rng;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::secp256k1::SecretKey;
+use bitcoin::secp256k1::SECP256K1;
+
+#[derive(Clone)]
+pub struct AesCipher {
+    secret_key: SecretKey,
+    inner: Aes256GcmSiv,
+}
+
+impl AesCipher {
```

Comment:
What's your confidence level on our encrypted backup scheme? I'm afraid I'm not sure if this is safe.

It's always a risk to roll your own crypto, so I would like to spend some more time figuring out if what we are doing here is safe. Other than the choice of cipher which LGTM, I'm not sure about sending over the nonce in plaintext to the server, to then retrieve it and use it when decrypting. That seems incorrect.

Reply:
> I'm not sure about sending over the nonce in plaintext to the server, to then retrieve it and use it when decrypting. That seems incorrect.

I based my implementation on that comment, suggesting that its fine to send the nonce in plaintext

> So you can either have each side keep track of a counter as the nonce, or you can send it in the clear before each message; so long as it's unique.

https://crypto.stackexchange.com/a/5818

Reply:
Here is the article I based my decision on adding the nonce in front of the message.

https://www.linkedin.com/pulse/understanding-nonces-use-aes-gcm-derek-hinch/

Reply:
Thanks for the info provided. This gives me much more confidence in the approach.

I know one of the sources mentions not needing the nonce to be random, but it seems safer to keep it random. And if we are gonna keep it random I think it would make sense to use something other than `thread_rng`.